lib/resty/dns/resolver.lua

```diff
@@ -797,6 +802,70 @@ function _M.compress_ipv6_addr(addr)
 end
 
 
+local function _expand_ipv6_addr(addr)
+    if find(addr, "::") then
+        local ncol = 0
+
+        for i=1, #addr do
+            if sub(addr, i, i) == ":" then
+                ncol = ncol + 1
+            end
+        end
+
+        if addr == "::" then
+            addr = "0::0"
+        elseif sub(addr, 1, 2) == "::" then
+            addr = "0" .. addr
+        end
+
+        addr = re_sub(addr, "::", ":" .. rep("0:",8 - ncol), "jo")
+    end
+
+    return addr
+end
+
+
+function _M.expand_ipv6_addr = _expand_ipv6_addr
+
+
+function _M.arpa_str(addr)
+    local arpa = ".in-addr.arpa"
+    
+    if find(addr, ":") then
+        addr, arpa = _expand_ipv6_addr(addr), ".ip6.arpa"
+        local tmp, idx, hidx, addrlen = new_tab(32,0), 1, 1, #addr
+        
+        for i = addrlen, 1, -1 do
+            local s = sub(addr, i, i)
```

Comment:
I don't think so, because we need to store the substring in the `tmp` table anyways if isn't a colon. If we use `string.byte` then we also have to use `string.char` later on.